src/commands/registration/pasv.js

```diff
@@ -8,15 +8,18 @@ module.exports = {
       return this.reply(502);
     }
 
+    let port;
     this.connector = new PassiveConnector(this);
     return this.connector.setupServer()
     .then((server) => {
-      let address = this.server.options.pasv_url;
+      port = server.address().port;
```

Comment:
This line can probably stay how it was `const { port } = server.address()` - you can keep it here. I would return a tuple from the promise `{address, port}` to avoid having to scope it outside of the function.

```suggestion
      const { port } = server.address();
```

Reply:
But then it would be up to the pasv_url function to return a port also, which is outside of its control. I also don't like moving port to the upper scope. I can also chain an internal .then to the pasv_url function so that port is known

Reply:
I think we are on the same page. I am thinking you just need to keep it within scope here, and you can use the same line as it was before. Remember, we want to avoid confusion between the `server.address()` function, and the pasv function.

Reply:
just commited the changes. Not sure I understood your request for pasvAddress naming. Let me know if that's what you meant.